### ISSUE TITLE
App frequently closing after update to 1.0.1

### ISSUE THREAD

**Issue by machiav3lli** [NeoApplications/Neo-Store#610]
### Discussed in https://github.com/NeoApplications/Neo-Store/discussions/607

<div type='discussions-op-text'>

<sup>Originally posted by **gbakeman** May 15, 2024</sup>
Just after updating to version 1.0.1 recently, the app has started closing frequently. Immediately after the update it closed as expected, then it would close while trying to update other apps as well. I left it alone but noticed it was periodically sending notifications indicating it was trying to update, so I force-closed it. Now I'm trying to open it, and it closes within one or two seconds. When manually looking through logs, I can see these lines that looks interesting:

```
05-15 12:55:14.232 E/PackageManager(  958): Attempt to invoke virtual method 'boolean android.os.Bundle.getBoolean(java.lang.String, boolean)' on a null object reference
05-15 12:55:14.232 I/ActivityManager(  958): Force stopping com.machiav3lli.fdroid appid=10122 user=-1: installPackageLI
```

I also attach an extended log here. Thanks in advance for any help!

[neostore-log.txt](https://github.com/NeoApplications/Neo-Store/files/15324839/neostore-log.txt)

Edit: I tried clearing the cache only which has allowed me to start using the app, however now it is spamming me with a notification that says "Installed" and nothing else. It self-clears then adds its self back again, even if I manually clear it. Force stopping the app has halted the notifications for now.</div>

**Comment by machiav3lli** [comment]
Can you try to get a crashlog using [Scoop](https://www.f-droid.org/de/packages/com.machiav3lli.fdroid/)? Also some helpful info would be:
- Selected installer
- Android version
- Any Magisk/Xposed modules?
- Updated 0.9.15 → 1.0.0 → 1.0.1 or jumped some versions?

**Comment by gbakeman** [comment]
Unfortunately, I'm no longer able to reproduce the crashing issue. I was able to successfully update an app through Neo Store without issue. That resulted in an expected "_app_ Installed" notification, however the blank "Installed" popups persist. And I apologize for omitting the basic information! 😄

- Installer: Root installer
- Android version: 9.0
- Magisk @ 27.0 w/ Always Trust User Certificates, MagiskHide Props Config, Systemless Hosts, Universal SafetyNet Fix, and Zygisk - LSPosed (XPosed framework) @ v1.9.2
  - Disable FLAG_SECURE, Scoop and XPrivacyLua modules installed.
- I believe I followed the upgrade path you've specified above.

**Comment by machiav3lli** [comment]
Someusers had some issues with XPrivacyLua or hiding some apps (mainly Magisk related) before, so this maybe the cause of the crash (as it seems an injected "coverup" to an expected function call.

On the black-installed-notification thingy, I'm still not sure why it's so persistent… I can only guess that there's a work running, but even then it should be ended when notifying "installed"… did you try to clean install? maybe export extras (if you have any favorites and app preferences you'd like to keep) and repositories (if you have added any repositories that aren't by default in NS)

**Comment by gbakeman** [comment]
I just want to add that exporting settings, clearing all data, and reimporting settings seems to have fixed the remaining notification issue. Trying to install apps while it was still crashing probably caused that. Unfortunately it's only detected a small number of apps that I have installed compared to before (should've exported the installed apps too, didn't think I needed to do that.) Do you still want to keep this open?

Thank you for all of your help on this.

**Comment by machiav3lli** [comment]
I would suggest using another mirror than the main for the official f-droid repo (I for once use FAU's, based in Germany)

**Comment by machiav3lli** [comment]
@gbakeman I'll keep it for now, maybe we (with @opusforlife2 ) can guess the reason behind the issue?!

> in my case the app crashed just 1 second after opening, so I didn't have the chance to do anything else within it.

That's surely an uncommon thing to happen… auto-fixing by clearing cache…

So here's my first guess: A question for you both, does these devices have (real) external storage (e.g. sd-card)?

**Comment by gbakeman** [comment]
> So here's my first guess: A question for you both, does these devices have (real) external storage (e.g. sd-card)?

I honestly can't remember. I've looked through my settings, and I see no mention of external storage! I think everything, at least for NeoStore, is stored on phone storage. I'm sorry the log wasn't helpful, hopefully more people will come here.

If you don't mind my asking, why do you recommend against the official F-Droid repo?

**Comment by machiav3lli** [comment]
> I honestly can't remember. I've looked through my settings

So your device doesn't have an sd-card? So maybe it was only an API-change thingy (as many APIs were changed lately)

> why do you recommend against the official F-Droid repo?

I don't recommend against the main repo, but I suggest using one of their mirrors (maybe the closest to your place, for better performance). This reduces the overload on the main mirror, as I guess this was the reason the sync failed for you at the start…

**Comment by machiav3lli** [comment]
But this is not really possible considering that no key have really changed in between, also cache shouldn't be causing such issue on 1.0.0→1.0.1, if so that should've happend already on 0.9.15→1.0.0

**Comment by machiav3lli** [comment]
The changes you mentioned are a 0.9.15→1.0.0 thing, and didn't come with 1.0.1. There was also no changes in the navigation items, so that also a default wouldn't break things… sometimes it's just some system caching magic…

**Comment by gbakeman** [comment]
Unfortunately, the crashing and superuser toast popups have returned. As I was typing this, I also just received a notification that a new app update is available. I was able to open Neo-Store by tapping the notification, and then install the app update. I was then able to use Neo-Store without issue. I can also open it again without crashing now. However, I am still seeing superuser toasts. It seems like it was struggling to work through something, but eventually completely what it was doing? Either way, it seems like clearing data hasn't fixed it.

**Comment by machiav3lli** [comment]
> Edit: Scoop, for some reason, is not catching these app closures as crashes.

I guess there's no real crashes (assuming you've setup Scoop after the last system restart).

> I have also seen these far more frequently than before, despite not using the app. Not sure what the app is doing here.

Do you have the option to keep app updated notification after installation? Root calls - when root installer is the one selected - are only used in installing and uninstalling apps. That said, I guess depending on how the root checking method is implemented, magisk could interpret it as a call instead of a check. The root toasts are showing up on launches or syncs?

**Comment by gbakeman** [comment]
> I guess there's no real crashes (assuming you've setup Scoop after the last system restart).

Correct, Scoop has caught a few other apps so I believe it's working.

> Do you have the option to keep app updated notification after installation? Root calls - when root installer is the one selected - are only used in installing and uninstalling apps. That said, I guess depending on how the root checking method is implemented, magisk could interpret it as a call instead of a check. The root toasts are showing up on launches or syncs?

It looks like I had that option turned off. At the time, the root toasts were showing repeatedly before and after I installed the app update. After I made that post, I left it alone, which caused my phone to be quite warm throughout the day and drain the battery. Force stopping the app and enabling battery optimization seemed to help with that, although there's likely an underlying problem.

![Screenshot_20240525-153713](https://github.com/NeoApplications/Neo-Store/assets/842942/8475443b-b720-42c2-9c90-95ebcca68e9d)
